language-plutus-core/exe/Main.hs

```diff
@@ -250,29 +253,37 @@ getProg inp fmt =
                return (fakeAlexPosn <$ plc)  -- Adjust the return type to ParsedProgram
                    where fakeAlexPosn = PLC.AlexPn 0 0 0
 
+
 ---------------- Typechecking ----------------
 
 runTypecheck :: TypecheckOptions -> IO ()
-runTypecheck (TypecheckOptions inp fmt) = do
-    prog <- getProg inp fmt
-    let cfg = PLC.defConfig
-    case PLC.runQuoteT $ PLC.typecheckPipeline cfg prog of
-      Left (e :: PlcParserError) -> do
-            T.putStrLn $ PLC.displayPlcDef e
-            exitFailure
-      Right ty -> do
-            T.putStrLn $ PLC.displayPlcDef ty
-            exitSuccess
+runTypecheck (TypecheckOptions inp fmt) =
+    case PLC.runQuoteT $ PLC.getStringBuiltinTypes () of
```

Comment:
Why not just put this inside the existing call to `runQuoteT`? i.e. just before doing `typecheckPipeline`

Reply:
I'm not sure what you mean.  If you're talking about the call to `getStringTypes`, that may fail, so you have to process the result to deal with `Error` or extract the meanings before you can feed them to `typeCheckPipeline`.

Reply:
Yes, but it throws a type error, no? So can't we just run it together with the typechecking function since they should both run in some error monad. Like so: https://github.com/input-output-hk/plutus/blob/master/plutus-tx-plugin/src/Language/PlutusTx/Plugin.hs#L318

Reply:
OK, I'll try that later.